apps/api/src/modules/quiz/services/duplicate-question.quiz.service.ts

```diff
@@ -0,0 +1,125 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { QuizQuestion as QuizQuestionEntity } from '../domain/quizzes_questions.entity';
+
+import { IDuplicateQuestionQuizService } from '../interfaces/services/duplicate-question.quiz.service.interface';
+import { DuplicateQuestionQuizDto } from '../dto/duplicate-question.quiz.dto';
+import { Explanation, Question } from 'src/modules/question/domain';
+import { QuestionTranslation } from 'src/modules/translation/domain/questionTranslation.entity';
+import { ExplanationTranslation } from 'src/modules/translation/domain/explanationTranslation.entity';
+import { Language } from 'src/modules/languages/domain';
+import { App } from 'src/modules/app/domain';
+import { QuestionImage } from 'src/modules/question_image/domain/question_images.entity';
+
+
+@Injectable()
+export class DuplicateQuestionQuizService implements IDuplicateQuestionQuizService{
+
+  constructor(
+    @InjectRepository(QuizQuestionEntity)
+    private readonly quizQuestionRepo: Repository<QuizQuestionEntity>,
+    @InjectRepository(Question)
+    private readonly questionRepo: Repository<Question>,    
+    @InjectRepository(App)
+    private readonly appRepo: Repository<App>,
+    @InjectRepository(Explanation)
+    private readonly explanationRepo: Repository<Explanation>,
+    @InjectRepository(QuestionTranslation)
+    private readonly questionTranslationRepo: Repository<QuestionTranslation>,
+    @InjectRepository(ExplanationTranslation)
+    private readonly explanationTranslationRepo: Repository<ExplanationTranslation>,
+    @InjectRepository(Language)
+    private readonly languageRepo: Repository<Language>,
+    @InjectRepository(QuestionImage)
+    private readonly questionImageRepo: Repository<QuestionImage>
+  ) {}
+
+  async execute (duplicateQuestionDto: DuplicateQuestionQuizDto) {
+    console.log("🚀 ~ DuplicateQuestionQuizService ~ execute ~ duplicateQuestionDto:", duplicateQuestionDto)
+    
+    const originalQuestion = await this.questionRepo.findOne({
+      where: { id: duplicateQuestionDto.questionId },
+      relations: [
+        'apps', 
+        'explanations', 
+        'questionTranslations', 
+        'images',
+        'explanations.explanationTranslations'
+      ],
+    });
+
+    if (!originalQuestion) {
+      throw new Error('Question not found');
+    }
+
+    console.log("🚀 ~ originalQuestion.languageId:", originalQuestion.languageId);
+
+    // Get default language if original doesn't have one
+    const language = originalQuestion.languageId 
+      ? await this.languageRepo.findOne({ where: { id: originalQuestion.languageId } })
+      : await this.languageRepo.findOne({ where: { code: 'en' } });
+
+    if (!language) {
+      throw new Error('Language not found');
+    }
+
+    const newQuestion = new Question();
+    newQuestion.name = `Copy ${originalQuestion.name}`;
```

Comment:
suggestion based on yannchwen's comment on mattermost ❤️ 
```suggestion
    newQuestion.name = `Copy of ${originalQuestion.name}`;
```